flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/service/FlinkService.java

```diff
@@ -168,4 +169,18 @@ public void stopSessionCluster(
         FlinkUtils.deleteCluster(deployment, kubernetesClient, deleteHaData);
         FlinkUtils.waitForClusterShutdown(kubernetesClient, conf);
     }
+
+    public void triggerSavepoint(FlinkDeployment deployment, Configuration conf) throws Exception {
+        LOG.info("Triggering savepoint on " + deployment.getMetadata().getName());
+        try (ClusterClient<String> clusterClient = getClusterClient(conf)) {
+            CompletableFuture<String> result =
+                    clusterClient.triggerSavepoint(
+                            JobID.fromHexString(deployment.getStatus().getJobStatus().getJobId()),
+                            null);
+            deployment.getStatus().getJobStatus().setSavepointLocation(result.get());
```

Comment:
Does this need to check the result is valid?

Reply:
fair point, I'll look into this

Reply:
for the time being, the job gets into an error state if it fails to create the checkpoint:

```reconciliationStatus:
    error: 'java.util.concurrent.ExecutionException: org.apache.flink.runtime.checkpoint.CheckpointException:
      Checkpoint triggering task Source: Custom Source (1/2) of job 00000000000000000000000000000000
      is not being executed at the moment. Aborting checkpoint. Failure reason: Not
      all required tasks are currently running.'
```

Reply:
fair point though we could add a more sophisticated error handling

Reply:
@morhidi, it's better to add the check of the result and log the failure information for checkpoint failure.

Reply:
@SteNicholas what kind of check do you think is relevant here on the result returned? We could wrap the call in a try-catch, but in this case the user won't get any feedback on the failed savepoint operation. @gyfora @wangyang0918 what do you think? Shall we introduce a new Savepoint object instead of the current savepointLocation field to track the result, error, timestamp etc?

Reply:
Yes I think we could have a Savepoint/SavepointInfo object in the `lastSavepoint` field of the jobstatus to add metadata like creation time that could be valuable

Reply:
If you want you can do that as part of the ticket, or as a follow up work I dont mind either way :)

Reply:
Let's do this in a separate ticket. It affects other parts of the code too cancel with savepoint, restore from savepoint. etc.

Reply:
Yes, we should capture the last successful savepoint (or maybe n) in the status. I think we should also capture the error as event on the CR,  not just the log.

Reply:
If we count savepoint failures as reconcile errors, it will be stored there. It's a nice user experience, once you remove the savepoint annotation the reconcile loop starts again.

Reply:
Is there a reason not to do that? Since the user expects the savepoint to be taken I think we should treat it as error if it fails.

Reply:
I also agree with counting savepoint failures as reconcile errors.